datalad/tests/utils_cached_dataset.py

```diff
@@ -0,0 +1,233 @@
+"""Utils for cached test datasets"""
+
+from datalad import cfg
+from datalad.core.distributed.clone import (
+    Clone,
+)
+from datalad.distribution.dataset import Dataset
+from datalad.utils import (
+    better_wraps,
+    ensure_list,
+    optional_args,
+    Path,
+    rmtree
+)
+from datalad.support.annexrepo import AnnexRepo
+from datalad.tests.utils import with_tempfile
+
+
+DATALAD_TESTS_CACHE = cfg.obtain("datalad.tests.cache")
+
+
+def url2filename(url):
+    """generate file/directory name from a URL"""
+
+    # TODO: Not really important for now, but there should be a more
+    #       sophisticated approach to replace. May be just everything that
+    #       isn't alphanumeric? Or simply hash the URL?
```

Comment:
It can be a big deal to have an additional clone. The immediate use case, causing this entire PR, is about a dataset with several container images. Getting that content is what I need a cache for.

But I do like a hash as well. Given that I don't want a clone per version, I'd prob. need to parse the URL and hash it w/o any parameters.